x-pack/plugins/actions/server/lib/action_executor.ts

```diff
@@ -415,14 +391,8 @@ async function getActionInfoInternal<Source = unknown>(
     );
   }
 
-  const actionsClient = await getActionsClientWithRequest(request, source);
-
-  // if not pre-configured action, should be a saved object
-  // ensure user can read the action before processing
-  const { actionTypeId, config, name } = await actionsClient.get({ id: actionId });
-
   const {
-    attributes: { secrets },
+    attributes: { secrets, actionTypeId, config, name },
```

Comment:
This is the main change of the PR. The RBAC check happened within the `await getActionsClientWithRequest` call. Instead of using `actionsClient.get` to get a few fields, I combined it with the `encryptedSavedObjectsClient.getDecryptedAsInternalUser` call below so it's only one request (bonus: perf improvement!).